### ISSUE TITLE
🐛 [firebase_messaging] onMessage called twice ONLY for MacOS

### ISSUE THREAD

**Issue by mlb5000** [firebase/flutterfire#11857]
## Bug report

When the app is in focus in the foreground on MacOS (testing on 13.4), the `FirebaseMessaging.onMessage.listen` handler is called twice.

### Steps to reproduce

The following code should reproduce the issue.

```
@override
void initState() {
    print('initState');

    startupAsync();
}

void startupAsync() async {
    print('startupAsync');
    await Firebase.initializeApp();

    FirebaseMessaging.onMessageOpenedApp.listen((message) {
      processNotification(message.data);
    });

    FirebaseMessaging.onMessage.listen((RemoteMessage message) {
          print('App in foreground notification');
    });
}
```

The print logs then look like this on startup, indicating we're only calling it once.

```
flutter: initState
flutter: startupAsync
```

Then when I send a _single_ notification, it calls the listener twice

```
flutter: App in foreground notification
flutter: App in foreground notification
```

The exact same codebase running on iOS does not have this issue.

### Expected behavior

The listener should only be called once.

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.7.12, on macOS 13.4 22F66 darwin-arm64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.1)
[✓] VS Code (version 1.84.1)
[✓] Connected device (2 available)
[✓] HTTP Host Availability

• No issues found!
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
- _flutterfire_internals 1.3.12 [collection firebase_core firebase_core_platform_interface flutter meta]
- archive 3.3.2 [crypto path]
- args 2.4.0
- asn1lib 1.4.0
- async 2.10.0 [collection meta]
- audioplayers_android 2.0.0 [audioplayers_platform_interface flutter]
- audioplayers_darwin 3.0.1 [audioplayers_platform_interface flutter]
- audioplayers_linux 1.0.4 [audioplayers_platform_interface flutter]
- audioplayers_platform_interface 4.0.0 [flutter plugin_platform_interface]
- audioplayers_web 2.2.0 [audioplayers_platform_interface flutter flutter_web_plugins js]
- audioplayers_windows 1.1.3 [audioplayers_platform_interface flutter]
- base32 2.1.3
- boolean_selector 2.1.1 [source_span string_scanner]
- cached_network_image_platform_interface 1.0.0 [flutter flutter_cache_manager]
- cached_network_image_web 1.0.1 [flutter flutter_cache_manager cached_network_image_platform_interface]
- characters 1.2.1
- clock 1.1.1
- collection 1.17.0
- convert 3.1.1 [typed_data]
- cross_file 0.3.3+4 [js meta]
- crypto 3.0.2 [typed_data]
- csslib 0.17.2 [source_span]
- dbus 0.7.3 [args ffi meta xml]
- fake_async 1.3.1 [clock collection]
- ffi 1.2.1
- file 6.1.4 [meta path]
- firebase_auth_platform_interface 7.0.4 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 5.8.7 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser js meta]
- firebase_core 2.22.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_core_platform_interface 5.0.0 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 2.8.1 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_messaging_platform_interface 4.5.13 [_flutterfire_internals firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 3.5.13 [_flutterfire_internals firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins js meta]
- flutter_blurhash 0.7.0 [flutter]
- flutter_cache_manager 3.3.0 [clock collection file flutter http path path_provider pedantic rxdart sqflite uuid]
- flutter_driver 0.0.0 [file flutter flutter_test fuchsia_remote_debug_protocol path meta vm_service webdriver archive async boolean_selector characters clock collection crypto js matcher material_color_utilities platform process source_span stack_trace stream_channel string_scanner sync_http term_glyph test_api typed_data vector_math]
- flutter_inappwebview_internal_annotations 1.1.0
- flutter_local_notifications_linux 0.5.1 [flutter flutter_local_notifications_platform_interface dbus path xdg_directories]
- flutter_local_notifications_platform_interface 5.0.0 [flutter plugin_platform_interface]
- flutter_sequence_animation 3.0.1 [flutter meta]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- freezed_annotation 1.1.0 [collection json_annotation meta]
- fuchsia_remote_debug_protocol 0.0.0 [process vm_service file meta path platform]
- hotp 0.0.2 [base32 crypto]
- http2 2.0.1
- http_parser 4.0.2 [collection source_span string_scanner typed_data]
- in_app_review_platform_interface 2.0.3 [flutter url_launcher plugin_platform_interface platform]
- js 0.6.5 [meta]
- json_annotation 4.8.0 [meta]
- lints 1.0.1
- matcher 0.12.13 [meta stack_trace]
- material_color_utilities 0.2.0
- meta 1.8.0
- mime 1.0.4
- octo_image 1.0.2 [flutter flutter_blurhash]
- package_info_plus_linux 1.0.5 [package_info_plus_platform_interface flutter path]
- package_info_plus_macos 1.3.0 [flutter]
- package_info_plus_platform_interface 1.0.2 [flutter meta plugin_platform_interface]
- package_info_plus_web 1.0.6 [flutter flutter_web_plugins http meta package_info_plus_platform_interface]
- package_info_plus_windows 1.0.5 [package_info_plus_platform_interface ffi flutter win32]
- path 1.8.2
- path_drawing 0.5.1+1 [vector_math meta path_parsing flutter]
- path_parsing 0.2.1 [vector_math meta]
- path_provider_android 2.0.24 [flutter path_provider_platform_interface]
- path_provider_foundation 2.2.1 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.10 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_platform_interface 2.0.6 [flutter platform plugin_platform_interface]
- path_provider_windows 2.0.7 [ffi flutter path path_provider_platform_interface win32]
- pedantic 1.11.1
- petitparser 5.1.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.4 [meta]
- pointer_interceptor 0.9.3+4 [flutter]
- pointycastle 3.7.2 [collection convert js]
- process 4.2.4 [file path platform]
- quiver 2.1.5 [matcher meta]
- sentry 6.22.0 [http meta stack_trace uuid intl]
- share_plus_linux 3.0.0 [share_plus_platform_interface file flutter meta url_launcher]
- share_plus_macos 3.0.1 [share_plus_platform_interface flutter]
- share_plus_platform_interface 3.2.0 [cross_file flutter meta mime plugin_platform_interface path_provider uuid]
- share_plus_web 3.0.0 [share_plus_platform_interface url_launcher flutter flutter_web_plugins meta]
- share_plus_windows 3.0.0 [share_plus_platform_interface flutter meta url_launcher]
- shared_preferences_android 2.1.0 [flutter shared_preferences_platform_interface]
- shared_preferences_foundation 2.2.0 [flutter shared_preferences_platform_interface]
- shared_preferences_linux 2.2.0 [file flutter path path_provider_linux path_provider_platform_interface shared_preferences_platform_interface]
- shared_preferences_platform_interface 2.2.0 [flutter plugin_platform_interface]
- shared_preferences_web 2.1.0 [flutter flutter_web_plugins shared_preferences_platform_interface]
- shared_preferences_windows 2.2.0 [file flutter path path_provider_platform_interface path_provider_windows shared_preferences_platform_interface]
- sign_in_with_apple_platform_interface 1.0.0 [flutter plugin_platform_interface meta]
- sign_in_with_apple_web 1.0.1 [flutter flutter_web_plugins sign_in_with_apple_platform_interface js]
- sky_engine 0.0.99
- source_span 1.9.1 [collection path term_glyph]
- sqflite 2.2.6 [flutter sqflite_common path]
- sqflite_common 2.4.3 [synchronized path meta]
- stack_trace 1.11.0 [path]
- stream_channel 2.1.1 [async]
- stream_mixin 3.0.4 [flutter]
- string_scanner 1.2.0 [source_span]
- sync_http 0.3.1
- synchronized 3.0.1
- term_glyph 1.2.1
- test_api 0.4.16 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- timezone 0.8.0 [path]
- typed_data 1.3.1 [collection]
- uni_links_platform_interface 1.0.0 [flutter plugin_platform_interface]
- uni_links_web 0.1.0 [flutter flutter_web_plugins uni_links_platform_interface]
- url_launcher_linux 2.0.3 [flutter]
- url_launcher_macos 2.0.3 [flutter]
- url_launcher_platform_interface 2.1.2 [flutter plugin_platform_interface]
- url_launcher_web 2.0.16 [flutter flutter_web_plugins url_launcher_platform_interface]
- url_launcher_windows 2.0.2 [flutter]
- vector_math 2.1.4
- vm_service 9.4.0
- webdriver 3.0.1 [archive matcher path stack_trace sync_http]
- webview_flutter_android 2.10.4 [flutter webview_flutter_platform_interface]
- webview_flutter_platform_interface 1.9.5 [flutter meta plugin_platform_interface]
- webview_flutter_wkwebview 2.9.5 [flutter path webview_flutter_platform_interface]
- win32 2.6.1 [ffi]
- xdg_directories 0.2.0+3 [meta path process]
- xml 5.4.1 [collection meta petitparser]
```

</details>

---

**Comment by danagbemava-nc** [comment]
Hi @mlb5000, can you provide a complete minimal reproducible code sample in a repo that we can clone so that we can investigate this? I have been trying to run the plugin example app but it can't run. 

Also, can you try the example app to see if you still face this issue?

**Comment by mlb5000** [comment]
Hey,

The issue here was that I moved some code over from the startup of my iOS application into the macOS side. The equivalent of this iOS code on macOS causes `onMessage` to be called multiple times, but has no negative effects on iOS.

```
    if #available(iOS 10.0, *) {
        UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
    }
```

**Comment by danagbemava-nc** [comment]
Hi @mlb5000, can you check with the plugin example app to see if you still get this issue?
